splash_ingest/server/poller.py

```diff
@@ -48,10 +52,31 @@ def init_logging():
 ingest_db = MongoClient(INGEST_DB_URI)[INGEST_DB_NAME]
 
 init_ingest_service(ingest_db, databroker_db)
+
+
+class TerminateRequested():
+    state = False
+
+
+terminate_requested = TerminateRequested
+
 poll_for_new_jobs(
     POLLER_SLEEP_SECONDS,
     SCICAT_BASEURL,
     SCICAT_INGEST_USER,
     SCICAT_INGEST_PASSWORD,
+    terminate_requested,
     THUMBS_ROOT
 )
+
+
+def sigterm_handler(signum, frame):
+    logger.info("sigterm received")
+    terminate_requested.state = True
+
+
+signal.signal(signal.SIGTERM, sigterm_handler)
```

Comment:
Given that this is a managed service and not a library, this is probably a fine this to do. But I'll mention that my impulse is usually to do this in a context manager, so that there is a way to put the original one back in charge when we are done. That only really applies if you are sharing the process with someone else's library/application.

Example: https://github.com/bluesky/bluesky/blob/566dce9e7ada95e2346c9aa61206c6a2f0a189bf/bluesky/utils/__init__.py#L159-L213

Reply:
Thanks. I think that's probably more than I need here, but that reference may come in handy!